ArduPlane/Plane.cpp

```diff
@@ -32,3 +32,31 @@ Plane::Plane(void)
 
 Plane plane;
 AP_Vehicle& vehicle = plane;
+
+#if AP_SCRIPTING_ENABLED
+
+// returns true if vehicle is landing. Only used by Lua scripts
+bool Plane::is_landing() const
+{
+    #if HAL_QUADPLANE_ENABLED
+    
+    return plane.quadplane.in_vtol_land_descent();
+
+    #endif // HAL_QUADPLANE_ENABLED
+
+    return control_mode->is_landing();
+}
+
+// returns true if vehicle is taking off. Only used by Lua scripts
+bool Plane::is_taking_off() const
+{
+    #if HAL_QUADPLANE_ENABLED
+
+    return plane.quadplane.in_vtol_takeoff();
+
+    #endif // HAL_QUADPLANE_ENABLED
+
+    return control_mode->is_taking_off();
```

Comment:
with quadplane enabled, this function has a double return which is unreachable code so please fix that. Also, watch out for whitespace on the # if, make it match with the rest of the code. This also needs to be moved to Arduplane.cpp under set_land_descent_rate() because there's already a group of functions there wrapped in AP_SCRIPTING_ENABLED

```suggestion
#if HAL_QUADPLANE_ENABLED
    if (plane.quadplane.in_vtol_takeoff()) {
        return true;
    }
#endif

    return control_mode->is_taking_off();
```

Reply:
```suggestion
    return control_mode->is_taking_off() || (started_flying_ms > 0 && millis() - started_flying_ms < 30000);
```

Reply:
>     return control_mode->is_taking_off() || (started_flying_ms > 0 && millis() - started_flying_ms < 30000);

Why 30 seconds? could/should this be a parameter?

Reply:
meh. Doesn't need to be. I think after 30 seconds of flying you're no longer considered taking off. What number would you like?

Reply:
> > ```
> > return control_mode->is_taking_off() || (started_flying_ms > 0 && millis() - started_flying_ms < 30000);
> > ```
> 
> Why 30 seconds? could/should this be a parameter?

@timtuxworth If it was a parameter, what time duration would you use?

Reply:
seconds. Default to 30 is fine, but depending on the size and capabilities of the vehicle or even the flying location, you might want it to be shorter or longer. Some might want longer to be absolutely sure it's really flying, some want to make it much shorter depending on what they are doing.

Reply:
I think I prefer the original patch. If were not sure from the flight mode specific handling then assume were not taking off. There is potential for `started_flying_ms` to be wrong.

Reply:
fair enough.. so just remove the 30s check for now? We can iterate on improvements later

Reply:
I see in TAKEOFF mode it's hardcoded to switch from flight_stage "TAKEOFF" to "NORMAL after s(tarted_flying_ms > 10000 && groundspeed > 3)

Reply:
I've removed the timer, it's using Flight Stage now per @IamPete1 's suggestion